### ISSUE TITLE
HeadlessUI not working with React18 and NextJS12.1

### ISSUE THREAD

**Issue by Stancobridge** [tailwindlabs/headlessui#1298]
**What package within Headless UI are you using?**

@headlessui/react

**What version of that package are you using?**

v1.5.0

**What browser are you using?**

Chrome, Safari, or N/A

**Reproduction URL**
am working on a private repo
**Describe your issue**

Headless UI stopped working once I upgraded to react@18 and next@12.1 but works when I run build the nextjs app with yarn build and run yarn start on the same react@18 and next@12.1

**Comment by joeblau** [comment]
Next 12.4.1 works with react 17.0.x. Seems like something with the new version of React.

**Comment by marcin-kopanski** [comment]
i had issues with Transition in React 18, but with 17 works perfect

**Comment by vvscode** [comment]
Same on me =)

**Comment by naufalfarras** [comment]
yes confirmed, added compatibility with latest react would be appreciate!

**Comment by sgamma** [comment]
For my case removing the <React.StrictMode> where app bootstrap fixed the Transition component.

**Comment by jlarmstrongiv** [comment]
See https://github.com/tailwindlabs/headlessui/issues/681

**Comment by joshxfi** [comment]
for `nextjs`, you can disable `reactStrictMode` on `next.config.js` and `headlessui` will work
```js
const nextConfig = {
   reactStrictMode: true,
}
```
or just downgrade to `react@17.0.x`

**Comment by kb1995** [comment]
I have a problem when I wrap Transition around a Popover. @joshxfi solution works quite well, but would appreciate a way to use Script Mode together with headlessUI

**Comment by RobinMalfait** [comment]
Hey! Thank you for your bug report!
Much appreciated! 🙏

As far as I can tell the biggest issue is the `Transition` component, all others seem to work as expected.
Going to close this as a duplicate of #681, but will be working on a fix for this. For now as a workaround, you should be able to remove `StrictMode`.

**Comment by atonb117** [comment]
FYI `reactStrictMode: false` is not a consistent solution. Gonna try downgrading from React 18

**Comment by RobinMalfait** [comment]
If you land on this issue, please go to #681 where we explain how to get the fixes installed for React 18 compatibility.
